src/backend/distributed/planner/local_distributed_join_planner.c

```diff
@@ -384,7 +384,7 @@ static bool
 HasConstantFilterOnUniqueColumn(RangeTblEntry *rangeTableEntry,
 								RelationRestriction *relationRestriction)
 {
-	if (rangeTableEntry == NULL)
+	if (rangeTableEntry == NULL || relationRestriction == NULL)
```

Comment:
might be useful to have a comment on the meaning of `relationRestriction == NULL`

Reply:
Just noting from our internal chat: In a sense, it doesn't make sense to recursively plan a table that doesn't contribute to the result. Still, for the time being it makes sense to not have any special cases in the logic as it might have some side-effect like the bug that we are fixing.

Reply:
actually, it might be nice to have a comment like the above in the code